src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,197 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Your index was: " + index);
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+        } else if (index < size / 2) {
+            Node addedNode = getByIndexForwards(index);
+            link(addedNode, value);
+            size++;
+        } else if (index >= size / 2) {
+            Node addedNode = getByIndexBackwards(index);
+            link(addedNode, value);
+            size++;
+        }
```

Comment:
You can use `return` instead of first else-if to improve readability, also you can extract some common logic (like node creation or size incrementation) out of the else-if blocks